### ISSUE TITLE
Compile error on latest `master` branch.

### ISSUE THREAD

**Issue by IceCodeNew** [shadowsocks/shadowsocks-rust#353]
rustc could not compile `shadowsocks-service` due to following errors: E0277, E0599.
Related commit: https://github.com/shadowsocks/shadowsocks-rust/commit/3a1054acc54cfc23a1cd232aa0a45b66af4bd612

Here is a complete error message emit by rustc:
<details>
<summary>compilation log：</summary>

```
error[E0277]: the trait bound `ProxyHttpStream: tokio::io::async_read::AsyncRead` is not satisfied --> crates/shadowsocks-service/src/local/http/server.rs:51:47 | 51 | let bypass_client = Client::builder().build::<_, Body>(BypassConnector::new(self.context.clone())); | ^^^^^ the trait `tokio::io::async_read::AsyncRead` is not implemented for `ProxyHttpStream` | = note: required because of the requirements on the impl of `hyper::client::connect::Connect` for `BypassConnector`

error[E0277]: the trait bound `ProxyHttpStream: tokio::io::async_write::AsyncWrite` is not satisfied --> crates/shadowsocks-service/src/local/http/server.rs:51:47 | 51 | let bypass_client = Client::builder().build::<_, Body>(BypassConnector::new(self.context.clone())); | ^^^^^ the trait `tokio::io::async_write::AsyncWrite` is not implemented for `ProxyHttpStream` | = note: required because of the requirements on the impl of `hyper::client::connect::Connect` for `BypassConnector`

error[E0277]: the trait bound `Upgraded: tokio::io::AsyncRead` is not satisfied --> crates/shadowsocks-service/src/local/http/dispatcher.rs:111:74 | 111 | let (mut plain_reader, mut plain_writer) = split(upgraded); | ^^^^^^^^ the trait `tokio::io::AsyncRead` is not implemented for `Upgraded` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.3.6/src/io/split.rs:36:12 | 36 | T: AsyncRead + AsyncWrite, | --------- required by this bound in `tokio::io::split`

error[E0277]: the trait bound `Upgraded: tokio::io::AsyncWrite` is not satisfied --> crates/shadowsocks-service/src/local/http/dispatcher.rs:111:74 | 111 | let (mut plain_reader, mut plain_writer) = split(upgraded); | ^^^^^^^^ the trait `tokio::io::AsyncWrite` is not implemented for `Upgraded` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.3.6/src/io/split.rs:36:24 | 36 | T: AsyncRead + AsyncWrite, | ---------- required by this bound in `tokio::io::split`

error[E0277]: the trait bound `Upgraded: tokio::io::AsyncRead` is not satisfied --> crates/shadowsocks-service/src/local/http/dispatcher.rs:114:33 | 114 | let _ = establish_tcp_tunnel( | ^^^^^^^^^^^^^^^^^^^^ the trait `tokio::io::AsyncRead` is not implemented for `Upgraded` | ::: crates/shadowsocks-service/src/local/utils.rs:21:14 | 21 | pub async fn establish_tcp_tunnel<PR, PW, SR, SW>( | -------------------- required by a bound in this ... 31 | PR: AsyncRead + Unpin, | --------- required by this bound in `establish_tcp_tunnel` | = note: required because of the requirements on the impl of `tokio::io::AsyncRead` for `tokio::io::ReadHalf<Upgraded>`

error[E0277]: the trait bound `Upgraded: tokio::io::AsyncWrite` is not satisfied --> crates/shadowsocks-service/src/local/http/dispatcher.rs:114:33 | 114 | let _ = establish_tcp_tunnel( | ^^^^^^^^^^^^^^^^^^^^ the trait `tokio::io::AsyncWrite` is not implemented for `Upgraded` | ::: crates/shadowsocks-service/src/local/utils.rs:21:14 | 21 | pub async fn establish_tcp_tunnel<PR, PW, SR, SW>( | -------------------- required by a bound in this ... 32 | PW: AsyncWrite + Unpin, | ---------- required by this bound in `establish_tcp_tunnel` | = note: required because of the requirements on the impl of `tokio::io::AsyncWrite` for `tokio::io::WriteHalf<Upgraded>`

error[E0599]: no method named `request` found for struct `hyper::Client<BypassConnector>` in the current scope --> crates/shadowsocks-service/src/local/http/dispatcher.rs:157:42 | 157 | match self.bypass_client.request(self.req).await { | ^^^^^^^ method not found in `hyper::Client<BypassConnector>` | ::: crates/shadowsocks-service/src/local/http/connector.rs:22:1 | 22 | pub struct BypassConnector { | -------------------------- doesn't satisfy `BypassConnector: hyper::client::connect::Connect` | = note: the method `request` exists but the following trait bounds were not satisfied: `BypassConnector: hyper::client::connect::Connect`

error[E0599]: no method named `request` found for struct `hyper::Client<ProxyConnector>` in the current scope --> crates/shadowsocks-service/src/local/http/dispatcher.rs:179:30 | 179 | match client.request(self.req).await { | ^^^^^^^ method not found in `hyper::Client<ProxyConnector>` | ::: crates/shadowsocks-service/src/local/http/connector.rs:89:1 | 89 | pub struct ProxyConnector { | ------------------------- doesn't satisfy `ProxyConnector: hyper::client::connect::Connect` | = note: the method `request` exists but the following trait bounds were not satisfied: `ProxyConnector: hyper::client::connect::Connect`

error[E0277]: the trait bound `ProxyHttpStream: tokio::io::async_read::AsyncRead` is not satisfied --> crates/shadowsocks-service/src/local/http/client_cache.rs:37:40 | 37 | let client = Client::builder().build::<_, Body>(ProxyConnector::new(self.context.clone(), server.clone())); | ^^^^^ the trait `tokio::io::async_read::AsyncRead` is not implemented for `ProxyHttpStream` | = note: required because of the requirements on the impl of `hyper::client::connect::Connect` for `ProxyConnector`

error[E0277]: the trait bound `ProxyHttpStream: tokio::io::async_write::AsyncWrite` is not satisfied --> crates/shadowsocks-service/src/local/http/client_cache.rs:37:40 | 37 | let client = Client::builder().build::<_, Body>(ProxyConnector::new(self.context.clone(), server.clone())); | ^^^^^ the trait `tokio::io::async_write::AsyncWrite` is not implemented for `ProxyHttpStream` | = note: required because of the requirements on the impl of `hyper::client::connect::Connect` for `ProxyConnector`

error[E0599]: no method named `poll` found for struct `Pin<&mut tokio_rustls::Accept<AddrStream>>` in the current scope --> crates/shadowsocks-service/src/local/http/http_tls/rustls.rs:38:55 | 38 | match ready!(Pin::new(accept_fut).poll($cx)) { | ^^^^ method not found in `Pin<&mut tokio_rustls::Accept<AddrStream>>` ... 58 | forward_stream_method!(self, cx, poll_read, buf) | ------------------------------------------------ in this macro invocation | ::: /usr/local/cargo/git/checkouts/hyper-817b53a166fc1c58/3d6bdbe/src/server/tcp.rs:234:5 | 234 | pub struct AddrStream { | --------------------- | | | doesn't satisfy `AddrStream: tokio::io::AsyncRead` | doesn't satisfy `AddrStream: tokio::io::AsyncWrite` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-rustls-0.21.1/src/lib.rs:135:1 | 135 | pub struct Accept<IO>(MidHandshake<server::TlsStream<IO>>); | ----------------------------------------------------------- doesn't satisfy `tokio_rustls::Accept<AddrStream>: futures::Future` | = note: the method `poll` exists but the following trait bounds were not satisfied: `AddrStream: tokio::io::AsyncRead` which is required by `tokio_rustls::Accept<AddrStream>: futures::Future` `AddrStream: tokio::io::AsyncWrite` which is required by `tokio_rustls::Accept<AddrStream>: futures::Future` = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `poll_read` found for struct `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` in the current scope --> crates/shadowsocks-service/src/local/http/http_tls/rustls.rs:58:42 | 58 | forward_stream_method!(self, cx, poll_read, buf) | ^^^^^^^^^ method not found in `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` | ::: /usr/local/cargo/git/checkouts/hyper-817b53a166fc1c58/3d6bdbe/src/server/tcp.rs:234:5 | 234 | pub struct AddrStream { | --------------------- | | | doesn't satisfy `AddrStream: tokio::io::AsyncRead` | doesn't satisfy `AddrStream: tokio::io::AsyncWrite` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-rustls-0.21.1/src/server.rs:8:1 | 8 | pub struct TlsStream<IO> { | ------------------------ doesn't satisfy `_: tokio::io::AsyncRead` | = note: the method `poll_read` exists but the following trait bounds were not satisfied: `AddrStream: tokio::io::AsyncRead` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncRead` `AddrStream: tokio::io::AsyncWrite` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncRead`

error[E0599]: no method named `poll` found for struct `Pin<&mut tokio_rustls::Accept<AddrStream>>` in the current scope --> crates/shadowsocks-service/src/local/http/http_tls/rustls.rs:38:55 | 38 | match ready!(Pin::new(accept_fut).poll($cx)) { | ^^^^ method not found in `Pin<&mut tokio_rustls::Accept<AddrStream>>` ... 64 | forward_stream_method!(self, cx, poll_write, buf) | ------------------------------------------------- in this macro invocation | ::: /usr/local/cargo/git/checkouts/hyper-817b53a166fc1c58/3d6bdbe/src/server/tcp.rs:234:5 | 234 | pub struct AddrStream { | --------------------- | | | doesn't satisfy `AddrStream: tokio::io::AsyncRead` | doesn't satisfy `AddrStream: tokio::io::AsyncWrite` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-rustls-0.21.1/src/lib.rs:135:1 | 135 | pub struct Accept<IO>(MidHandshake<server::TlsStream<IO>>); | ----------------------------------------------------------- doesn't satisfy `tokio_rustls::Accept<AddrStream>: futures::Future` | = note: the method `poll` exists but the following trait bounds were not satisfied: `AddrStream: tokio::io::AsyncRead` which is required by `tokio_rustls::Accept<AddrStream>: futures::Future` `AddrStream: tokio::io::AsyncWrite` which is required by `tokio_rustls::Accept<AddrStream>: futures::Future` = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `poll_write` found for struct `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` in the current scope --> crates/shadowsocks-service/src/local/http/http_tls/rustls.rs:64:42 | 64 | forward_stream_method!(self, cx, poll_write, buf) | ^^^^^^^^^^ method not found in `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` | ::: /usr/local/cargo/git/checkouts/hyper-817b53a166fc1c58/3d6bdbe/src/server/tcp.rs:234:5 | 234 | pub struct AddrStream { | --------------------- | | | doesn't satisfy `AddrStream: tokio::io::AsyncRead` | doesn't satisfy `AddrStream: tokio::io::AsyncWrite` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-rustls-0.21.1/src/server.rs:8:1 | 8 | pub struct TlsStream<IO> { | ------------------------ doesn't satisfy `_: tokio::io::AsyncWrite` | = note: the method `poll_write` exists but the following trait bounds were not satisfied: `AddrStream: tokio::io::AsyncRead` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncWrite` `AddrStream: tokio::io::AsyncWrite` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncWrite`

error[E0599]: no method named `poll_flush` found for struct `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` in the current scope --> crates/shadowsocks-service/src/local/http/http_tls/rustls.rs:71:75 | 71 | TlsStreamState::Streaming(ref mut stream) => Pin::new(stream).poll_flush(cx), | ^^^^^^^^^^ method not found in `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` | ::: /usr/local/cargo/git/checkouts/hyper-817b53a166fc1c58/3d6bdbe/src/server/tcp.rs:234:5 | 234 | pub struct AddrStream { | --------------------- | | | doesn't satisfy `AddrStream: tokio::io::AsyncRead` | doesn't satisfy `AddrStream: tokio::io::AsyncWrite` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-rustls-0.21.1/src/server.rs:8:1 | 8 | pub struct TlsStream<IO> { | ------------------------ doesn't satisfy `_: tokio::io::AsyncWrite` | = note: the method `poll_flush` exists but the following trait bounds were not satisfied: `AddrStream: tokio::io::AsyncRead` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncWrite` `AddrStream: tokio::io::AsyncWrite` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncWrite`

error[E0599]: no method named `poll_shutdown` found for struct `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` in the current scope --> crates/shadowsocks-service/src/local/http/http_tls/rustls.rs:79:75 | 79 | TlsStreamState::Streaming(ref mut stream) => Pin::new(stream).poll_shutdown(cx), | ^^^^^^^^^^^^^ method not found in `Pin<&mut tokio_rustls::server::TlsStream<AddrStream>>` | ::: /usr/local/cargo/git/checkouts/hyper-817b53a166fc1c58/3d6bdbe/src/server/tcp.rs:234:5 | 234 | pub struct AddrStream { | --------------------- | | | doesn't satisfy `AddrStream: tokio::io::AsyncRead` | doesn't satisfy `AddrStream: tokio::io::AsyncWrite` | ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-rustls-0.21.1/src/server.rs:8:1 | 8 | pub struct TlsStream<IO> { | ------------------------ doesn't satisfy `_: tokio::io::AsyncWrite` | = note: the method `poll_shutdown` exists but the following trait bounds were not satisfied: `AddrStream: tokio::io::AsyncRead` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncWrite` `AddrStream: tokio::io::AsyncWrite` which is required by `tokio_rustls::server::TlsStream<AddrStream>: tokio::io::AsyncWrite`

warning: unused import: `future::Future` --> crates/shadowsocks-service/src/local/http/http_tls/rustls.rs:4:5 | 4 | future::Future, | ^^^^^^^^^^^^^^ | = note: `#[warn(unused_imports)]` on by default

error: aborting due to 16 previous errors; 1 warning emitted

Some errors have detailed explanations: E0277, E0599.

For more information about an error, try `rustc --explain E0277`.

error:
failed to compile `shadowsocks-rust v1.9.0 (https://github.com/shadowsocks/shadowsocks-rust.git#3a1054ac)`, intermediate artifacts can be found at `/tmp/cargo-installRYIvl5`

Caused by:
could not compile `shadowsocks-service`

Caused by:
process didn't exit successfully: `rustc --crate-name shadowsocks_service --edition=2018 crates/shadowsocks-service/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C panic=abort -C linker-plugin-lto -C codegen-units=1 --cfg 'feature="http"' --cfg 'feature="hyper"' --cfg 'feature="local"' --cfg 'feature="local-dns"' --cfg 'feature="local-http"' --cfg 'feature="local-http-rustls"' --cfg 'feature="local-redir"' --cfg 'feature="local-socks4"' --cfg 'feature="local-tunnel"' --cfg 'feature="manager"' --cfg 'feature="mio"' --cfg 'feature="rand"' --cfg 'feature="rustls-native-certs"' --cfg 'feature="server"' --cfg 'feature="strum"' --cfg 'feature="strum_macros"' --cfg 'feature="tokio-rustls"' --cfg 'feature="tower"' --cfg 'feature="trust-dns"' --cfg 'feature="trust-dns-proto"' --cfg 'feature="trust-dns-resolver"' --cfg 'feature="webpki-roots"' -C metadata=37d97b5e3062c478 -C extra-filename=-37d97b5e3062c478 --out-dir /tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/cargo-installRYIvl5/release/deps --extern async_trait=/tmp/cargo-installRYIvl5/release/deps/libasync_trait-9526d89cabf636d3.so --extern byte_string=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libbyte_string-8c7d6cdd52def470.rmeta --extern byteorder=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libbyteorder-62a510dc4fe28f85.rmeta --extern bytes=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libbytes-3b0019b14b71c879.rmeta --extern cfg_if=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libcfg_if-2b222585ba722044.rmeta --extern futures=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libfutures-9965897e2a316d66.rmeta --extern http=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libhttp-797ddba49db1e9c2.rmeta --extern hyper=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libhyper-622493ad1bf043d4.rmeta --extern ipnet=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libipnet-1e59959dda5c75fc.rmeta --extern iprange=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libiprange-ad2f3f6793e663f7.rmeta --extern json5=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libjson5-5d5438c039a77b02.rmeta --extern lazy_static=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/liblazy_static-2ccedd180e114d21.rmeta --extern libc=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/liblibc-27f55a6cfdbacbbd.rmeta --extern log=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/liblog-49b810abaf91c4f4.rmeta --extern log4rs=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/liblog4rs-602717dc52d15246.rmeta --extern lru_time_cache=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/liblru_time_cache-814961fe6f7676e6.rmeta --extern mio=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libmio-7ebaf528cd602dbc.rmeta --extern pin_project=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libpin_project-c28f3abb2b3a1cb3.rmeta --extern qrcode=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libqrcode-c610aac389118572.rmeta --extern rand=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/librand-9684f183c24eb8da.rmeta --extern regex=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libregex-e86e75177affefc8.rmeta --extern rustls_native_certs=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/librustls_native_certs-1126b7608d78356d.rmeta --extern serde=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libserde-21a6e1bc6a4b4856.rmeta --extern shadowsocks=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libshadowsocks-44afa3a85ff74d3a.rmeta --extern socket2=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libsocket2-888e3f89384a077e.rmeta --extern spin=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libspin-f27a0deebc20a815.rmeta --extern strum=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libstrum-58819bcc9bf8e32d.rmeta --extern strum_macros=/tmp/cargo-installRYIvl5/release/deps/libstrum_macros-298d98ed17634535.so --extern thiserror=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libthiserror-e7f7e2ba5fa7e287.rmeta --extern tokio=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libtokio-b46430e2f1cb698d.rmeta --extern tokio_rustls=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libtokio_rustls-5b9f0897ad880300.rmeta --extern tower=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libtower-ce9af135c716f447.rmeta --extern trust_dns_proto=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libtrust_dns_proto-713bba1d5ff54c0a.rmeta --extern trust_dns_resolver=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libtrust_dns_resolver-5773a472f2d5ecf7.rmeta --extern webpki_roots=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/deps/libwebpki_roots-b66f756e4b97f282.rmeta -C target-feature=+crt-static -L native=/tmp/cargo-installRYIvl5/x86_64-unknown-linux-gnu/release/build/ring-ee0c00dbf75ec29e/out` (exit code: 1)
```

</details>

**Comment by IceCodeNew** [comment]
You may reproduce this error by following steps in this Dockerfile:
https://github.com/IceCodeNew/rust-collection/blob/0deea1907a38e3619f769d4baff2f187f8f4050b/Dockerfile#L1-L10

```
FROM quay.io/icecodenew/rust-collection:nightly_build_base_ubuntu AS shadowsocks-rust
SHELL ["/bin/bash", "-o", "pipefail", "-c"]
# https://api.github.com/repos/shadowsocks/shadowsocks-rust/commits?per_page=1
ARG shadowsocks_rust_latest_commit_hash=3a1054acc54cfc23a1cd232aa0a45b66af4bd612
RUN source '/root/.bashrc' \
    && RUSTFLAGS="-C target-feature=+crt-static" cargo install --bins -j "$(nproc)" --target x86_64-unknown-linux-gnu --no-default-features --features "logging trust-dns local server manager utility local-dns local-http local-http-rustls local-tunnel local-socks4 local-redir mimalloc" --git 'https://github.com/shadowsocks/shadowsocks-rust.git' --verbose \
    && cd /usr/local/cargo/bin || exit 1 \
    && strip sslocal ssmanager ssserver ssurl \
    && bsdtar -a -cf ss-rust-linux-gnu-x64.tar.xz sslocal ssmanager ssserver ssurl; \
    rm -f sslocal ssmanager ssserver ssurl
```

**Comment by zonyitoo** [comment]
`hyper` have been upgraded to tokio v1.0. Will be fixed by #354
